spec/requests/posts_controller_spec.rb

```diff
@@ -0,0 +1,31 @@
+require 'rails_helper'
+
+RSpec.describe PostsController, type: :controller do
+  let(:user) { User.create!(name: 'Test User', bio: 'Test bio.') }
+
+  describe 'GET #index' do
+    it 'assigns @user and @posts' do
+      user.posts.create(title: 'Post 1', text: 'This is post 1')
+      user.posts.create(title: 'Post 2', text: 'This is post 2')
+
+      get :index, params: { user_id: user.id }
+      expect(assigns(:user)).to eq(user)
+      expect(assigns(:posts)).not_to be_empty
+    end
+  end
+
+  describe 'GET #show' do
+    let(:post) { Post.create!(title: 'Test Post', text: 'This is a test post.', author: user) }
+
+    it 'assigns @user and @post' do
+      get :show, params: { user_id: user.id, id: post.id }
+      expect(assigns(:user)).to eq(user)
+      expect(assigns(:post)).to eq(post)
```

Comment:
* As stated in the [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/controller_specs_project.md#:~:text=Make%20sure%20that%20for,includes%20correct%20placeholder%20text.) that 👇 
```
Make sure that for each action you check:
If the response status was correct.
If a correct template was rendered.
If the response body includes the correct placeholder text.
```

- * Kindly ensure you write at least 3 tests that cover all scenarios stated above for each action